q2_amr/card/partition.py

```diff
@@ -98,3 +101,85 @@ def _collate(partition_list):
                     duplicate(file, collated_partitions.path / sample.name / file.name)
 
     return collated_partitions
+
+
+def partition_mags_annotations(
+    annotations: CARDAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def partition_reads_allele_annotations(
+    annotations: CARDAlleleAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDAlleleAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def partition_reads_gene_annotations(
+    annotations: CARDGeneAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDGeneAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def _partition_annotations(
+    annotations: Union[
+        CARDAnnotationDirectoryFormat,
+        CARDGeneAnnotationDirectoryFormat,
+        CARDAlleleAnnotationDirectoryFormat,
+    ],
+    num_partitions: int = None,
+):
+    partitioned_annotations = {}
+
+    # Get dict with paths to all files in artifact and get number of annotations
+    annotations_dict = annotations.sample_dict()
+    num_annotations = len(annotations_dict)
+
+    # If no number of partitions is specified or the number is higher than the number
+    # of annotations, all annotations get partitioned by annotation
+    if num_partitions is None:
+        num_partitions = num_annotations
+    elif num_partitions > num_annotations:
+        warnings.warn(
+            "You have requested a number of partitions"
+            f" '{num_partitions}' that is greater than your number"
+            f" of annotations '{num_annotations}'. Your data will be"
+            f" partitioned by annotation into '{num_annotations}'"
+            " partitions."
+        )
+        num_partitions = num_annotations
+
+    # Split the dict into a list of specified number of dicts
+    i = itertools.cycle(range(num_partitions))
+    annotations_split_list = [{} for _ in range(num_partitions)]
+    for key, value in annotations_dict.items():
+        annotations_split_list[next(i)][key] = value
```

Comment:
Hi @misialq 
I tried to use the array_split function of numpy but because my tuples have multiple paths in them and not only one path like in the other partition methods, it does not work.
I get this error: `ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (3, 3) + inhomogeneous part.`

I have now created a small helper function called split list that splits the list into a number of sublists.